### ISSUE TITLE
[Bug] Map editor cannot load maps set to 4 players

### ISSUE THREAD

**Issue by SilPho** [CorsixTH/CorsixTH#2223]
### Describe the issue

When trying to load a map that has a player count of "4 players", the game will throw an error and prevent you from editing that map file:
`Error while loading game: Lua\game_ui.lua:79: Player index out of range: 1`

This doesn't seem to cause problems if the player count is set to 1, 2, or 3, nor does it occur if you change the player count away from 4 before saving the file.

This means the file is unusable, which could ruin hours of work spent on making a map.

### Steps to Reproduce

1. Open the map editor
2. Set the player count to 4 using the top menu
3. Save the map with any name you like
4. Immediately try to load that map file. You don't need to leave the editor

### Expected Behaviour

The map can be opened and edited as normal.

### Save Game

Save game attached to report: Yes/No

### CorsixTH Version

0.66

### Operating System

Windows 10

### Theme Hospital Version

Origin

**Comment by lewri** [comment]
Looks like error comes from this:
https://github.com/CorsixTH/CorsixTH/blob/81db3338d89df253e85fc6b7ace9d937092dafaa/CorsixTH/Src/th_lua_map.cpp#L388-L399

**Comment by Alberth289346** [comment]
Map specification: https://github.com/CorsixTH/theme-hospital-spec/blob/master/format-specification.md#level  
Says byte 0 is number of players.
``` sh
$ xxd 0.map | head -2
00000000: 0400 0000 0000 0000 0000 0000 0000 0000  ................
00000010: 0000 0000 0000 0000 0000 0000 0000 0000  ................
```

``` diff
 bool level_map::get_player_camera_tile(int iPlayer, int* pX, int* pY) const {
+  printf("iPlayer = %d, get_player_count()=%d\n", iPlayer, get_player_count());
   if (iPlayer < 0 || iPlayer >= get_player_count()) {
```
Loading map in map editor:
```
Created game with savegame version 172.       # Displayed by just opening the editor -> ie default map
iPlayer = 0, get_player_count()=1
Created game with savegame version 172.       # Displayed when loading the map file, player_count is wrong.
iPlayer = 0, get_player_count()=0
<LocalisedString> Current value:Error while loading game: ...h/corsix-th/corsix_alberth_fork/CorsixTH/Lua/game_ui.lua:79: Player index out of range: 1

```

**Comment by lewri** [comment]
Fix won't hit until 0.67 release (for those not building from master -- may be worth noting in the wiki page that setting player count to 4 breaks the map.

**Comment by SilPho** [comment]
No problem. I'll adjust the wiki now
